087-external-certificate-manager.md

```diff
@@ -0,0 +1,324 @@
+
+## Motivation
+
+Strimzi's primary purpose is to provide a way to run Apache Kafka clusters on Kubernetes.
+Although it is nice that it can issue certificates, it would be beneficial if the certificates could be issued by a dedicated certificate manager, such as [cert-manager](https://cert-manager.io/).
+This is a feature that is often requested, especially because many organizations have specific compliance requirements with regard to certificates, for example:
+* Requiring that CA private keys are not shared.
+* Requiring that self-signed certificates cannot be used.
+
+## Proposal
+
+Strimzi will be updated to allow users to specify that certificates should be issued by an external certificate manager instead of issued by Strimzi.
+This proposal will specifically describe how this would work for cert-manager, however the user API for configuration will be written in a way that does not prevent other external certificate managers being added in the future.
+
+The proposal makes a few assumptions:
+* Strimzi will not be responsible for installing cert-manager, but we will document the supported versions of cert-manager that we have tested with.
+* Strimzi will not be responsible for creating `Issuer` or `ClusterIssuer` custom resources.
+* Strimzi will not be responsible for requesting a CA certificate be issued, it will only interact with cert-manager to request end-entity certificates are issued.
+* Strimzi will create `Certificate` custom resources (to request a new end-entity certificate is issued) and will allow the user to influence the contents of these resources by exposing options in the `Kafka` custom resource.
+* Strimzi will not directly interact with the lower level `CertificateRequest` and `CertificateSigningRequests` custom resources.
+* When Strimzi creates a `Certificate` custom resource, cert-manager will issue the certificate within a reasonable amount of time such that Strimzi can wait during the reconciliation.
+  If cert-manager is taking longer than expected Strimzi will fail the reconciliation and retry during the next reconciliation.
+* Users will provide to the Strimzi Cluster Operator the CA certificates it must trust for the current issuer via a Kubernetes Secret.
+
+### API
+
+The existing `spec.clusterCa` and `spec.clientsCa` fields will be extended to add a new property `type`:
+
+```yaml
+spec:
+  clusterCa:
+    validityDays: <integer> # notBefore=now, notAfter=now + validityDays
+    generateCertificateAuthority: <boolean>
+    generateSecretOwnerReference: <boolean>
+    renewalDays: <integer> # days before notAfter when we should start renewal
+    certificateExpirationPolicy: <renew-certificate|replace-key>
+    type: <strimzi.io|cert-manager.io> # (1)
+    certManagerIssuerRef: # (2)
+      name: <string>
+      kind: <Issuer|ClusterIssuer>
+      group: <string> # cert-manager.io by default
+    publicCert: # (3)
```

Comment:
I wonder if we can have some better name for this to make it more clea that this is the CA public certificate ...

Reply:
`publicCaCert` ?

Reply:
I think having the CA part is a good point, in future we might specify other certificates e.g. for calling a certificate manager API. I also get the point about a certificate is a public key, so using publicCert is not entirely accurate.

@scholzj @Frawless @ppatierno what about either `caCert` or `caPublicKey`? `caCert` aligns with the terminology we use in the docs today https://strimzi.io/docs/operators/latest/deploying#security-using-your-own-certificates-str but `caPublicKey` is more specific

Reply:
I am for `caCert` at this point.

Reply:
`caCert` sounds good to me.

Reply:
+1 for `caCert`

Reply:
Ok, updated to be `caCert`